### PR TITLE
DHFPROD-5147: Back-end and middle-tier merge step validation

### PR DESCRIPTION
**Pull request by ryanjdew**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/impl/sm-es-impl.xqy [hunk 1]
```diff
@@ -233,6 +234,7 @@ declare function es-impl:get-definition-properties(
               => map:with("pathExpression", $path-expression)
               => map:with("propertyPath", $property-path)
               => map:with("isIndexed", $is-indexed)
+              => map:with("allowsMultiple", $allows-multiple)
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/impl/sm-es-impl.xqy:237]: I would call this "allowsMultipleValues" to make it explicit what "multiple" refers to.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/validateMergingStep.api [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "functionName": "validateMerge",
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/validateMergingStep.api:2]: For consistency, this should be the same as the endpoint name.

Comment (rjrudin) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/validateMergingStep.api:2]: Also, should this PR include updating the MasteringService interface? Can run "gradle generatemasteringserviceinterface"

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/impl/sm-es-impl.xqy (hunk 1 above)
Comment (ryanjdew) [reply marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/impl/sm-es-impl.xqy]: renamed

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/validateMergingStep.api (hunk 2 above)
Comment (ryanjdew) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/validateMergingStep.api]: added the middle-tier logic

**Review by rjrudin — APPROVED** [review]

**Review by fsnow — APPROVED** [review]

**Review by rjrudin — APPROVED** [review]

**Comment by rjrudin** [comment]
@AkshayBajajML  @ryanjdew  Should this still be done for 5.5, given the new direction for the Jobs team?

**Review by fsnow — APPROVED** [review]

**Pull request merged**